analyses/phylo_beta_diversity/phylo_beta_diversity.py

```diff
@@ -478,3 +493,183 @@ def calculate_phylo_beta_diversity_sorensen(pam, tree):
         Matrix(phylo_beta_sne_data, headers=mtx_headers),
         Matrix(beta_sor_data, headers=mtx_headers),
         Matrix(phylo_beta_sor_data, headers=mtx_headers))
+
+
+# .............................................................................
+def calc_phylo_jac_distr(pam, tree, nrand = 5):
+    """Calculates distribution of jaccard metrics based on randomization 
+        of phylogenetic relationships.
+
+    Args:
+        pam (:obj:`Matrix`): A Lifemapper Matrix object with presence absence
+            values (site rows by species columns).
+        tree (:obj:`TreeWrapper`): A TreeWrapper object for a wrapped Dendropy
+            phylogenetic tree.
+
+    Returns:
+        Mean & SD of distribution of Jaccard-based metrics from randomizations.
+
+    Note:
+        * It looks like the scipy.spatial.distance.jaccard method may be useful
+            here.
+
+    Todo:
+        * Fill in method documentation
+        * Fill in method
+    """
+    # Get a lookup dictionary for the matrix index of each species in the PAM
+    #    in case they are not in the same order as the taxa in the tree
+    species_lookup = get_species_index_lookup(pam)
+
+    # Build a header dictionary, all of the returned matricies will have the
+    #    same headers, site rows by site columns.
+    # Note: This will differ from the R method because each site will be
+    #    present in both the rows and the columns.
+    mtx_headers = {
+        '0': pam.get_row_headers(),  # Row headers
+        '1': pam.get_row_headers()  # Column headers
+    }
+
+    num_sites = pam.data.shape[0]  # Get the number of sites in the PAM
+
+    # print pam.data, "\n"
+    # print pam.get_column_headers(),"\n"
+    # print pam.get_row_headers(),"\n"
+
+    # These matrices will serve as running average placeholders.
+    pjtu_av = np.zeros((num_sites, num_sites), dtype=np.float)
+    pjne_av = np.zeros((num_sites, num_sites), dtype=np.float)
+    pjac_av = np.zeros((num_sites, num_sites), dtype=np.float)
+
+    # TODO: Randomize tree, calc. metrics, save running average.
+    for trial in range(len(nrand)):

```

Comment:
```suggestion
    #for trial in range(len(nrand)):
```
Since the loop hasn't been filled in, comment it out